pkg/util/util.go

```diff
@@ -64,8 +68,37 @@ var (
 	lazyConfig *rest.Config
 	lazyRest   *rest.RESTClient
 	lazyClient client.Client
+
+	// InsecureHTTPTransport is a global insecure http client
+	InsecureHTTPTransport = &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	// SecureHTTPClient is a global secured http client
+	SecureHTTPTransport = newSecureHTTPTransport()
 )
 
+// newSecureHTTPTransport initializes a secured http transport. http.Transport may cause memory leaks when overused
+func newSecureHTTPTransport() *http.Transport {
+	// Load CA cert
+	caCert, err := ioutil.ReadFile(serviceCAPath)
+	if err != nil {
+		return &http.Transport{}
```

Comment:
Do we prefer `return nil` to make it use the default transport?

Reply:
so, returning nil there caused later on a runtime error, even if I checked it for nil and didn't assign anything to the client's transport. returning  &http.Transport{} worked. do you have other suggestion that will be ok ?

Reply:
I'm curious - what is the error when it's nil?

Reply:
Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)

Reply:
of course, but where? what is the stack trace?